docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,285 @@
+1. Have separate namespaced and clustered Ratify CRs. Cluster admin has the access to configure both cluster-wide and namespaced CRs. Team admins can only access namespaced CRs. Both CRs would look almost the same except for the scope. Users would have to maintain 2 sets of CRs but don't need to reserve a namespace for cluster-wide CRs.
+2. Have only namespaced Ratify CRs. For the cluster-wide CRs, we reserve a namespace for it, such as gatekeeper-system. And only cluster admin could access resources in reserved namespace. In this option, users only need to maintain a single CRD but have to keep a reserved namespace.
+
+## User Experience
+We've discussed several aspects concerning functionalities and implementation. However, we also need to consider the user experience when enabling multi-tenancy support.
+
+From the perspective of user experience, it makes a great difference between 2 deployment strategy.
+
+### Deployment per Namespace
+1. Cluster admin or team admin must deploy a Ratify service in each required namespace.
+2. Upon a new namespace created, cluster admin or team admin must deploy a Ratify service for it. Otherwise there should be some mechanism to automate it. Additionally, admin needs to apply new `Provider` CRs and update `Assign` CRs if individual control on tag mutation is required.
+3. And if customers want to have a different Redis instance for each namespace, they have to maintain the Redis deployment as well.
+4. Data isolation is more strict following this strategy. And customers have more granular control over Ratify configurations, such as cache config and log config.
+5. Users need to maintain their own TLS certificates for mTLS between Gatekeeper and Ratify.
+
+### Deployment per Cluster
+1. Cluster admin must deploy a Ratify service in the cluster.
+2. There is no need to deploy new Ratify service upon namespace creation. Admin only needs to update `Assign` CRs if individual control on tag mutation is required.
+3. If customers want to have a different Redis instance for each namespace, they have to maintain the Redis deployment as well.
+4. Data isolation is weaker following this strategy. And it has some limitation on the granularity of Ratify configurations, such as cache config and log config.
+5. Users don't need to manually maintain TLS certificates.
+
+# Proposed Solutions
+As we already discussed a few areas that needs to be considered for multi-tenancy, there are 2 general options to Ratify's use case. These 2 options are different in terms of how we deploy Ratify service in the cluster.
+
+
+## Ratify Deployment per Namespace or Cluster
+Current Ratify is deployed under gatekeeper-system namespace by default. In the multi-tenancy case, users could either deploy a Ratify service in each namespace or in a specific namespace, such as gatekeeper-system.
+
+Notes: a Ratify service includes all dependency service, such as Redis.
+
+An overview of deployment per namespace:
```

Comment:
Do we have detailed design on the CRD behavior change? i guess most of them will have namespaced level?

Reply:
this is under discussion, and I'll have a detailed design soon.